Java/core/src/main/java/com/amazon/randomcutforest/state/sampler/CompactSamplerState.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.state.sampler;
+
+import java.util.Arrays;
+
+import lombok.Data;
+
+import com.amazon.randomcutforest.sampler.CompactSampler;
+
+@Data
+public class CompactSamplerState {
+
+    public float[] weightArray;
```

Comment:
minor. the fields can be private with lombok.

Reply:
These fields are being used in code that will be deleted, and I don't want to add those changes to this PR. We'll have another chance to review all the data classes when we add them to the 2.0 branch.